### PR TITLE
Updates bgpv1 pkg to use netip.Addr type

### LINKED ISSUE #24246: Convert net.IP to netip.Addr
Meta issue to track the conversion of net.IP to netip.Addr in our code base

See the [steps to contribute](https://github.com/cilium/cilium/issues/24246#issuecomment-1500445741) to this issue below.

### PR DESCRIPTION
**Pull request by danehans**
Updates the bgpv1 pkg to use netip.Addr type.

Partially Fixes: #24246

### CODE REVIEW

File: pkg/node/address.go [hunk 1]
```diff
@@ -284,6 +293,11 @@ func GetIPv4() net.IP {
 	return clone(n.GetNodeIP(false))
 }
 
+func GetAddrV4() *netip.Addr {
+	n := localNode.Get()
+	return cloneAddr(n.GetNodeAddr(false))
+}
```
Comment (christarazi) [inline pkg/node/address.go:299]: Hm, I don't think it makes much sense to duplicate entire functions just to convert one package (BGP) over to the `netip.Addr` type. I think we'd prefer the entire `pkg/node` to be converted and then downstream packages from there.

I think this approach introduces unnecessary code churn in the tree without an immediate benefit, with an additional risk of the duplicated functions having bug(s) / regression(s).

**Review by christarazi — CHANGES_REQUESTED** [review]
Thanks for the PR!

A comment for rethinking the approach in this PR below.

File: pkg/node/address.go (hunk 1 above)
Comment (danehans) [reply pkg/node/address.go:299]: > Hm, I don't think it makes much sense to duplicate entire functions just to convert one package (BGP) over to the netip.Addr type. I think we'd prefer the entire pkg/node to be converted and then downstream packages from there.

I originally took this approach, see #25485 for details. However, changing the node pkg turned into a massive PR.

Comment (christarazi) [reply pkg/node/address.go:299]: Understood. I think the best way forward is to convert the `net.*` usages in the BGP pkg directly without having to create extra functions in other packages.

**Pull request closed without merging**